plugins/gnome-keyring.py

```diff
@@ -23,44 +23,47 @@
 gi.require_version('Peas', '1.0')
 gi.require_version('PeasGtk', '1.0')
 gi.require_version('Liferea', '3.0')
-gi.require_version('GnomeKeyring', '1.0')
-from gi.repository import GObject, Peas, PeasGtk, Gtk, Liferea, GnomeKeyring
+gi.require_version('Secret', '1')
+from gi.repository import GObject, Peas, PeasGtk, Gtk, Liferea, Secret
+
+LABEL = 'liferea'
+SCHEMA = Secret.Schema.new(
+    "net.sf.liferea.Secret",
+    Secret.SchemaFlags.NONE,
+    {'id': Secret.SchemaAttributeType.STRING}
+)
 
 class GnomeKeyringPlugin(GObject.Object, Liferea.AuthActivatable):
     __gtype_name__ = 'GnomeKeyringPlugin'
 
     object = GObject.property(type=GObject.Object)
 
-    def do_activate(self):
-        result = GnomeKeyring.unlock_sync("liferea", None)
-        if GnomeKeyring.Result.OK != result:
-            print("There doesn't seem to be a keyring yet.")
+    def get_collection(self):
+        service = Secret.Service.get_sync(Secret.ServiceFlags.LOAD_COLLECTIONS, None)
+        colls = service.get_collections()
+        for c in colls:
+            if c.get_label() == LABEL:
+                return c
+        return None
 
     def do_deactivate(self):
```

Comment:
Maybe this could be removed. It doesn't do anything.

Reply:
In do_query, one could use Service.search_sync instead of Collection.search_sync.
But in do_store, Item.create_sync relies on this `c` being the right collection (with the given label), which is not necessarily true for a user with more than one collection. So please keep this check in there. (Actually there can be more than one liferea collections if anyone else would be using this label for their collection, which I do not assume.)

If you would not care about any old liferea keyrings, you would make use of aliases, which give you a simplified API, but I thought backwards compatibility is better.

Reply:
I was talking about the `do_deactivate`.

Reply:
I did not touch it, because I don't know any details about the peas plugin system. Maybe it is triggered. If you like, you can remove it after merging.